pkg/controllers/mcs/mcs_serviceexport_controller.go

```diff
@@ -0,0 +1,235 @@
+	oldSvcExport := &mcsv1alpha1.ServiceExport{}
+	err := c.Client.Get(ctx, types.NamespacedName{Name: mcs.Name, Namespace: mcs.Namespace}, oldSvcExport)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+		newSvcExport := &mcsv1alpha1.ServiceExport{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: mcs.Namespace,
+				Name:      mcs.Name,
+				Annotations: map[string]string{
+					networkingv1alpha1.ResourceKindMultiMultiClusterServiceUsedBy: string(networkingv1alpha1.ExposureTypeCrossCluster),
+				},
+				OwnerReferences: []metav1.OwnerReference{
+					*metav1.NewControllerRef(ownSvc, corev1.SchemeGroupVersion.WithKind("Service")),
+				},
+			},
+		}
+		err = c.Client.Create(ctx, newSvcExport)
+		if err != nil {
+			msg := fmt.Sprintf("Create derived ServiceExport(%s/%s) failed, Error: %v", newSvcExport.Namespace, newSvcExport.Name, err)
+			klog.Errorf(msg)
+			c.EventRecorder.Eventf(mcs, corev1.EventTypeWarning, events.EventSyncDerivedServiceExportFailed, msg)
+			return err
+		}
+		msg := fmt.Sprintf("Sync the serviceExport of MultiClusterService(%s/%s) successful", mcs.Namespace, mcs.Name)
+		klog.V(4).Infof(msg)
+		c.EventRecorder.Eventf(mcs, corev1.EventTypeNormal, events.EventSyncDerivedServiceExportSucceed, msg)
+		return nil
+	}
+	if addCrossClusterToUsedByAnnotationIfNeed(oldSvcExport) {
+		oldSvcExport.OwnerReferences = append(oldSvcExport.OwnerReferences, *metav1.NewControllerRef(ownSvc, corev1.SchemeGroupVersion.WithKind("Service")))
+		err = c.Client.Update(ctx, oldSvcExport)
+		if err != nil {
+			msg := fmt.Sprintf("Update derived ServiceExport(%s/%s) failed, Error: %v", oldSvcExport.Namespace, oldSvcExport.Name, err)
+			klog.Errorf(msg)
+			c.EventRecorder.Eventf(mcs, corev1.EventTypeWarning, events.EventSyncDerivedServiceExportFailed, msg)
+			return err
+		}
+		msg := fmt.Sprintf("Sync the serviceExport of MultiClusterService(%s/%s) successful", mcs.Namespace, mcs.Name)
+		klog.V(4).Infof(msg)
+		c.EventRecorder.Eventf(mcs, corev1.EventTypeNormal, events.EventSyncDerivedServiceExportSucceed, msg)
+		return nil
+	}
+	return nil
+}
+
+func (c *MultiClusterServiceController) deleteDerivedServiceExport(ctx context.Context, mcs *networkingv1alpha1.MultiClusterService) error {
+	derivedSvcExport := &mcsv1alpha1.ServiceExport{}
+	derivedSvcExportNamespacedName := types.NamespacedName{
+		Namespace: mcs.Namespace,
+		Name:      mcs.Name,
+	}
+	err := c.Client.Get(ctx, derivedSvcExportNamespacedName, derivedSvcExport)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+	if removeCrossClusterToUsedByAnnotationIfNeed(derivedSvcExport) {
+		err = c.Client.Update(ctx, derivedSvcExport)
+	}
+	used := derivedSvcExport.Annotations[networkingv1alpha1.ResourceKindMultiMultiClusterServiceUsedBy]
+	if len(used) == 0 {
+		err = c.Client.Delete(ctx, derivedSvcExport)
+	}
+	if err != nil && !apierrors.IsNotFound(err) {
```

Comment:
Under what circumstances would `IsNotFound` occur?

Reply:
serviceExport has been manually deleted, but the informer has not yet synchronized.

Reply:
ok.